File: cap-test/deploy/config/deploy.rb
set :application, "cap-test"
set :repository,  "git@github.com:takkyuuplayer/sandbox.git"

role :localhost, "localhost"

set :scm, :git
set :branch, 'dev-capistrano_with_local'

set :use_sudo, false
set :group_writtable, true

set :deploy_via, :copy
set :copy_cache,      '/srv/work/cap-deployment/capcache'
set :copy_dir,        '/srv/work/cap-deployment/local-tmp'
set :deploy_to,       '/srv/work/cap-deployment/cap-test'
set :normalize_asset_timestamps, false

File: perl-ex/config/deploy.pl
use common::sense;
use Cinnamon::DSL;

set application => 'sandbox';
set repository  => 'git@github.com-tp:takkyuuplayer/sandbox.git';
set user        => 'vagrant';
set rsync_src   => '/tmp/cinnamon-deployment';

role
    vagrant => ['localhost', '127.0.0.1'],
    {
    deploy_to => '/srv/work/cinnamon-deployment',
    branch    => 'master',
    };

task rsync => sub {
    my ($host)    = @_;
    my $user      = get('user');
    my $rsync_src = get('rsync_src');
    my $deploy_to = get('deploy_to');
    run "rsync", "-az", $rsync_src, "$user\@$host:$deploy_to";
};

task deploy => sub {
    my ($host) = @_;
    call "rsync", $host;
};
